### PR TITLE
Update: support numeric-separator in no-loss-of-precision (refs #13568)

### PR DESCRIPTION
**Pull request by anikethsaha**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
Added support for `numeric separators` in `no-loss-of-precision`

#### Is there anything you'd like reviewers to focus on?
ref #13568

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Added "do not merge" as a reminder to update `package.json` before merging.

File: lib/rules/no-loss-of-precision.js
Comment (mdjermanovic) [inline lib/rules/no-loss-of-precision.js:59]: Maybe this function doesn't have to be changed, so we could avoid one extra `getRaw` call?

Numeric separators shouldn't affect the `node.raw.startsWith(prefix)` check.

As for the `!/^0[0-7]+$/u.test` condition, legacy octal literals can't have numeric separators.

Comment (anikethsaha) [reply lib/rules/no-loss-of-precision.js]: Thanks for the points,  having a separator there would be thrown by the parser as well.

**Review by mdjermanovic — CHANGES_REQUESTED** [review]
Looks good! Just one small jsdoc suggestion

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Comment by mdjermanovic** [comment]
#13609 has been merged, so this can be rebased now, and then we could remove "do not merge" label :)

File: lib/rules/no-loss-of-precision.js
Comment (mdjermanovic) [inline lib/rules/no-loss-of-precision.js:43]:
```suggestion
        /**
         * Gets the source code of the given number literal. Removes `_` numeric separators from the result.
         * @param {Node} node the number `Literal` node
         * @returns {string} raw source code of the literal, without numeric separators
         */
```

Seems this was lost in rebasing

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
